### PR TITLE
Start working on draft version 03.

### PR DESCRIPTION
**Pull request by EdOverflow**
https://github.com/securitytxt/security-txt/milestone/4

### CODE REVIEW

**Comment by austinheap** [comment]
@EdOverflow ABNF's should be <pre>'d I believe if you wanted to get that into this draft

File: draft-foudil-securitytxt.md [hunk 1]
```diff
@@ -317,6 +322,8 @@ directive.
 As stated in {{encryption}} and {{signature}}, both encryption keys
 and external signature files SHOULD be loaded over HTTPS.
 
+Web services MUST reserve the security.txt namespace to ensure no third-party can create a page with the "security.txt" name.
```
Comment (nightwatchcyber) [inline draft-foudil-securitytxt.md]: Can we explain what this means exactly? "web services" can have many meanings

Comment (nightwatchcyber) [inline draft-foudil-securitytxt.md]: Can we explain what this means? "Web services" has multiple meanings. Are we talking about a multi user host like Wordpress.com or Blogger?

File: draft-foudil-securitytxt.md [hunk 2]
```diff
@@ -112,17 +117,15 @@ HTTPS. Security email addresses SHOULD use the conventions defined
 in section 4 of {{!RFC2142}}, but there is no requirement for this directive
 to be an email address.
 
-While URIs already include the ability to have both email address and phone
-numbers via "mailto" and "tel" prefixes, allowing this information to be listed
-without a prefix is intended for ease of use and readability.
+The value MUST follow the general syntax described in {{!RFC3986}}. This means that "mailto" and "tel" URI schemes are required.
```
Comment (nightwatchcyber) [inline draft-foudil-securitytxt.md]: I wonder if the wording should be "This means that "mailto" and "tel" URI schemes **MUST be used when specifying email addresses and telephone numbers"**

File: draft-foudil-securitytxt.md [hunk 3]
```diff
@@ -226,7 +229,7 @@ Signature: https://example.com/.well-known/security.txt.sig
 
 ## Web-based services
 
-Web-based services SHOULD place the security.txt file under the /.well-known/ path; e.g. https://example.com/.well-known/security.txt.
+Web-based services SHOULD place the security.txt file under the /.well-known/ path; e.g. https://example.com/.well-known/security.txt. A security.txt file located under the top-level path SHOULD either redirect to the security.txt file under the /.well-known/ path or be used as a fall back.
```
Comment (nightwatchcyber) [inline draft-foudil-securitytxt.md]: Do we want to explain what "fall back" means? Same for line 69 above.

Perhaps "A security.txt file located under the top-level path SHOULD redirect to the security.txt file under the /.well-known/. **If it is not possible to place the security.txt file in the /.well-known/ path or setup a redirect, web-based services MAY place the file in the top-level path as a fall back option***"

File: draft-foudil-securitytxt.md [hunk 4]
```diff
@@ -257,11 +260,12 @@ implementors MUST ignore any fields they do not explicitly support.
 # File Format Description
 
 The expected file format of the security.txt file is plain text as defined
-in section 4.1.3 of {{!RFC2046}} and encoded in UTF-8.
+in section 4.1.3 of {{!RFC2046}} and follow {{!RFC5198}}.
```
Comment (nightwatchcyber) [inline draft-foudil-securitytxt.md]: I would recommend to follow the convention of RFC 7252:
https://tools.ietf.org/html/rfc7252#section-3.2

and would suggest the following wording: **"and is encoded using UTF-8 [RFC3629] in Net-Unicode form [RFC5198]"**

File: draft-foudil-securitytxt.md [hunk 5]
```diff
@@ -257,11 +260,12 @@ implementors MUST ignore any fields they do not explicitly support.
 # File Format Description
 
 The expected file format of the security.txt file is plain text as defined
```
Comment (nightwatchcyber) [inline draft-foudil-securitytxt.md]: Wondering also if we want to change "plain text" to read "**plain text (MIME type "text/plain")**"

File: draft-foudil-securitytxt.md [hunk 6]
```diff
@@ -140,7 +143,7 @@ Encryption: https://example.com/pgp-key.txt
 ## Signature: {#signature}
 
 In order to ensure the authenticty of the security.txt file one SHOULD use the
-"Signature:" directive, which allows you to link to an external signature. External signature files should be
+"Signature:" directive, which allows you to link to an external signature. External signature files SHOULD be
 named "security.txt.sig" and also be placed under the /.well-known/ path.
 External signature files SHOULD be loaded over HTTPS.
```
Comment (nightwatchcyber) [inline draft-foudil-securitytxt.md]: Do we want to add verbiage here about the possibility of an attacker subverting this, or leave that for the "Security Considerations" section?

File: draft-foudil-securitytxt.md [hunk 7]
```diff
@@ -70,8 +76,7 @@ can have an unlimited number of fields. It is important to note that
 you need a separate line for every field. One MUST NOT chain multiple
 values for a single directive. Everything MUST be in a separate field.
 
-A security.txt file only applies to the domain, subdomain, IPv4 or IPv6 address
-it is located in.
+A security.txt file only applies to the domain in the URI used to retrieve it, not to any of its subdomains or parent domains.
```
Comment (nightwatchcyber) [inline draft-foudil-securitytxt.md]: What about IP addresses? Only for that IP and not anything in the same subnet, above or below? Or is that self evident?

What about file systems?

File: draft-foudil-securitytxt.md [hunk 8]
```diff
@@ -55,12 +55,18 @@ In this document, the key words "MUST", "MUST NOT", "REQUIRED",
...
+Development of this draft takes place on Github at: https://github.com/securitytxt/security-txt
+
 # The Specification
 
-security.txt is a text file that should be located under the
+security.txt is a text file that SHOULD be located under the
 /.well-known/ path ("/.well-known/security.txt") {{!RFC5785}} for web
-properties. For file systems and version control repositories a .security.txt
-file should be placed in the root directory. This text file contains 4 directives
+properties. A security.txt file located under the website's top-level directory can either redirect to the security.txt file located under the /.well-known/ path or be used as a fall back. For web-based services, the instructions must be accessible via the Hypertext Transfer Protocol {{!RFC1945}} as a resource of Internet Media Type "text/plain;charset=utf8" {{!RFC1590}}. For file systems and version control repositories a .security.txt file SHOULD be placed in the root directory. This text file contains 4 directives
```
Comment (nightwatchcyber) [inline draft-foudil-securitytxt.md]: We may want a new line before "This text file contains..."
 
Also "contains 4 directives" isn't really true anymore, maybe reword this as "contains multiple directives"?

File: draft-foudil-securitytxt.md [hunk 9]
```diff
@@ -55,12 +55,18 @@ In this document, the key words "MUST", "MUST NOT", "REQUIRED",
...
 with different values. The "directive" is the first part of a field all the way up
```
Comment (nightwatchcyber) [inline draft-foudil-securitytxt.md]: Do we want to reference how headers are parsed in other protocols?

File: draft-foudil-securitytxt.md (hunk 7 above)
Comment (nightwatchcyber) [inline draft-foudil-securitytxt.md]: I also wonder if "only applies" may need to read strong as "MUST only apply"?

File: draft-foudil-securitytxt.md (hunk 8 above)
Comment (nightwatchcyber) [inline draft-foudil-securitytxt.md]: The reference to RFC1590 should probably be changed to point to section 4.1.3 of RFC2046 as per the rest of the draft, or maybe the media type registration.

For the charset, we may want to say something like "should be served with the default charset parameter set to "utf-8"

**Comment by EdOverflow** [comment]
@austinheap, the ABNF grammar is already `<pre>`'d.

File: draft-foudil-securitytxt.md (hunk 1 above)
Comment (EdOverflow) [reply draft-foudil-securitytxt.md]: @nightwatchcyber, is there an RFC that we could reference here in order to define https://en.wikipedia.org/wiki/Web_service?

Comment (nightwatchcyber) [reply draft-foudil-securitytxt.md]: Can we first figure out what we mean by "web service"? In my mind I was thinking of a domain name or IP addresses providing connections via HTTP protocol on port 80, but not necessarily machine to machine only. What is your thought?
